cmake/modules/kernel.cmake

```diff
@@ -0,0 +1,231 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2021, Nordic Semiconductor ASA
+
+# Zephyr Kernel CMake module.
+#
+# This is the main Zephyr Kernel module which is resposible for creation of
+# Zephyr libraries and the Zephyr executeable.
```

Comment:
Perhaps `kernel.cmake` is a bit misleading as we we often refer to a subset of the Zephyr repository codebase as the "kernel", whereas this CMake module is not specific to that part of the codebase?

Reply:
I tend to agree with you that it would be nice to have the kernel name referring to a specific sub-part of the Zephyr repo.
I'm very open to other proposals, but so far `kernel.cmake` was the best I could find that refers to the overall build of the system and the final executable.

We use the term `kernel` in a lot of places outside the kernel code base itself, for example:
The project name:
https://github.com/zephyrproject-rtos/zephyr/blob/5fda6a3b2fe90d7ace31657f5b64e4115ec75035/cmake/app/boilerplate.cmake#L591

https://github.com/zephyrproject-rtos/zephyr/blob/5fda6a3b2fe90d7ace31657f5b64e4115ec75035/cmake/app/boilerplate.cmake#L611

In Kconfig:
https://github.com/zephyrproject-rtos/zephyr/blob/5fda6a3b2fe90d7ace31657f5b64e4115ec75035/Kconfig.zephyr#L96-L103

https://github.com/zephyrproject-rtos/zephyr/blob/5fda6a3b2fe90d7ace31657f5b64e4115ec75035/Kconfig.zephyr#L109

https://github.com/zephyrproject-rtos/zephyr/blob/5fda6a3b2fe90d7ace31657f5b64e4115ec75035/Kconfig.zephyr#L361-L362

Because we are already using the term kernel to also refer to the overall project / image then I felt kernel was the best I could come up with, but other proposals are more than welcome.

Reply:
> Because we are already using the term kernel to also refer to the overall project / image then I felt kernel was the best I could come up with, but other proposals are more than welcome.

How about "core"?